Source/ACE.DatLoader/DatReader.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.IO;
+
+namespace ACE.DatLoader
+{
+    public class DatReader
+    {
+        public int Offset { get; set; }
+        private byte[] buffer;
+
+        public DatReader(string datFilePath, uint offset, uint size, uint sectorSize)

```

Comment:
I think I would rather see DatDatabase return a DatReader when given a file Id.  For example, in DatDatabase:
public DatReader GetReaderForFile(uint fileId)

The design paradigm here is that you've already loaded a DatDatabase, so you shouldn't have to go fishing around in objects to find the properties to pass into here.  DatDatabase already has them all, you just need to give it a file Id.